src/AdoNetCore.AseClient/AseParameter.cs

```diff
@@ -1,44 +1,212 @@
-﻿using System.Data;
+﻿using System;
+using System.Data;
+using System.Diagnostics;
 
 namespace AdoNetCore.AseClient
 {
     /// <summary>
     /// Represents a parameter to an <see cref="AseCommand" />. This class cannot be inherited.
     /// </summary>
+    [DebuggerDisplay("{" + nameof(ParameterName) + "}")]
     public sealed class AseParameter : IDbDataParameter
     {
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private string _parameterName;
 
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private AseDbType _type;
 
         /// <summary>
-        /// Gets or sets the <see cref="DbType" /> of the parameter.
+        /// Constructor function for an <see cref="AseParameter" />instance.
         /// </summary>
-        DbType IDataParameter.DbType 
-        { 
-            get 
+        public AseParameter()
+        {
+            Direction = ParameterDirection.Input;
+            SourceVersion = DataRowVersion.Default;
+            IsNullable = false;
+        }
+
+        /// <summary>
+        /// Constructor function for an <see cref="AseParameter" />instance.
+        /// </summary>
+        /// <param name="parameterName">The name of the parameter to add.</param>
+        /// <param name="value">An object that is the value of the parameter.</param>
+        public AseParameter(string parameterName, object value) : this()
+        {
+            Value = value ?? throw new ArgumentNullException();
+            ParameterName = parameterName;
+
+            if (value == DBNull.Value)
+            {
+                // Do nothing.
+            }
+            else if (value is int)
+            {
+                AseDbType = AseDbType.Integer;
+            }
+            else if (value is short)
+            {
+                AseDbType = AseDbType.SmallInt;
+            }
+            else if (value is long)
+            {
+                AseDbType = AseDbType.BigInt; // NOTE: - char is not supported by the SAP AseClient.
```

Comment:
Instead of
`AseDbType = AseDbType.BigInt; // NOTE: - char is not supported by the SAP AseClient.`
Should this be
`AseDbType = AseDbType.BigInt; // NOTE: - bigint is not supported by the SAP AseClient.`
or
`AseDbType = AseDbType.BigInt;`